deps/wazuh_testing/wazuh_testing/tools/monitoring.py

```diff
@@ -993,3 +994,32 @@ def wait_mtime(path, time_step=5, timeout=-1):
         if last_mtime - tic >= timeout:
             logger.error(f"{len(open(path, 'r').readlines())} lines within the file.")
             raise TimeoutError("Reached timeout.")
+
+
+def wait_until_agent_active( agent_id ):
+    con = sqlite3.connect('/var/ossec/queue/db/global.db')
+
+    cursor = con.cursor()
+    cursor.execute(f"select connection_status from agent where id={agent_id}")
+
+    agent_registered = -1
+
+    while agent_registered == -1:
+        try:
+            time.sleep(1)
+
+            cursor.execute(f"select connection_status from agent where id={agent_id}")
+
+            agent_status = cursor.fetchone()[0]
+
+            agent_registered = 0
+        except TypeError:
+            agent_registered = -1
+
+    while agent_status != 'active':
+            time.sleep(1)
+            cursor.execute(f"select connection_status from agent where id={agent_id}")
+            agent_status = cursor.fetchone()[0]
+
+    cursor.close()
+    con.close()
```

Comment:
This is duplicated from https://github.com/wazuh/wazuh-qa/blob/1024-remote-active-response/deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py#L585

Reply:
Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11